### PR TITLE
minor fixes: use actual tx id, prevent getting to nodeSetup if mining

### PR DESCRIPTION
**Pull request by yhaspel**
1. In Transaction, show actual Tx id
2. Prevent getting to nodeSetup if mining or in setup for mining
3. change password loader issue and clear verified password on cancel
4. update last backup time in settings page

### CODE REVIEW

File: app/screens/auth/CreateWallet.js [hunk 1]
```diff
@@ -192,14 +195,15 @@ class CreateWallet extends Component<Props, State> {
...
+  nextAction = async () => {
+    const { history, location, getMiningStatus } = this.props;
     const { subMode } = this.state;
     const isWalletOnlySetup = !!location?.state?.withoutNode;
+    const miningStatus = await getMiningStatus();
     if (subMode === 1 && this.validate()) {
       this.createWallet();
     } else if (subMode === 2) {
-      if (isWalletOnlySetup) {
+      if (isWalletOnlySetup || miningStatus !== nodeConsts.NOT_MINING) {
```
Comment (IlyaVi) [inline app/screens/auth/CreateWallet.js]: why is this here? Should it be merged to develop?

File: app/redux/node/actions.js [hunk 2]
```diff
@@ -30,8 +30,10 @@ export const getMiningStatus = (): Action => async (dispatch: Dispatch): Dispatc
   try {
     const status = await httpService.getMiningStatus();
     dispatch({ type: SET_MINING_STATUS, payload: { status } });
+    return status;
```
Comment (IlyaVi) [inline app/redux/node/actions.js]: what is the cause for this addition?

File: app/screens/auth/CreateWallet.js (hunk 1 above)
Comment (yhaspel) [reply app/screens/auth/CreateWallet.js]: We want to prevent a case in which we're mining or in the mining setup phase and also redirecting to the nodeSetup section. You suggested last time that we should not have that in the NodeSetup component, but where the redirection occurs, which is here.

File: app/redux/node/actions.js (hunk 2 above)
Comment (yhaspel) [reply app/redux/node/actions.js]: In create wallet 'nextAction' function, I need to check the status and not allow redirection to nodeSetup, as per last comment, if you agree with it.

File: app/screens/auth/CreateWallet.js (hunk 1 above)
Comment (IlyaVi) [reply app/screens/auth/CreateWallet.js]: Ok, then why do you make request for status in click handler? Can it be done before and wrapped by connection check to prevent unnecessary errors and changes in actions?

Comment (yhaspel) [reply app/screens/auth/CreateWallet.js]: What do you suggest? Can be done in componentDidMount for example. In this instance it's like wallet only setup.

Comment (IlyaVi) [reply app/screens/auth/CreateWallet.js]: Remove getMiningStatus from componentDidMount func in Main and move to healthCheckFlow in App. And rename healthCheckFlow

File: app/App.js [hunk 3]
```diff
@@ -122,12 +123,17 @@ class App extends React.Component<Props, State> {
...
-    this.healthCheckInterval = setInterval(() => {
-      store.dispatch(checkNodeConnection());
-    }, healthCheckIntervalTime);
+  connectionAndMiningFlow = async () => {
+    const connectionAndMiningIntervalTime = 360000;
+    const isConnected = await store.dispatch(checkNodeConnection());
+    isConnected && (await store.dispatch(getMiningStatus()));
+    this.connectionAndMiningInterval = setInterval(async () => {
+      const isConnected = await store.dispatch(checkNodeConnection());
+      const miningStatus = isConnected ? await store.dispatch(getMiningStatus()) : nodeConsts.NOT_MINING;
```
Comment (IlyaVi) [inline app/App.js]: no need to ask for mining status in loop, there is logic that asks for mining status only after mining was initiated

File: app/screens/main/Main.js [hunk 4]
```diff
@@ -231,21 +230,11 @@ class Main extends Component<Props, State> {
...
-  componentDidMount() {
-    const { isConnected, miningStatus, getMiningStatus } = this.props;
-    if (isConnected && miningStatus === nodeConsts.NOT_MINING) {
-      getMiningStatus();
-    }
-  }
-
   componentDidUpdate(prevProps: Props) {
-    const { isConnected, miningStatus, getMiningStatus, getGenesisTime } = this.props;
+    const { isConnected, miningStatus, getGenesisTime } = this.props;
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: return all logic inside componentDidUpdate back

File: app/screens/main/Main.js [hunk 5]
```diff
@@ -139,82 +139,82 @@ class Main extends Component<Props, State> {
         <Logo />
         <InnerWrapper>
           <NavBar>
-          <NavBarPart>
-            <NavLinksWrapper>
+            <NavBarPart>
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: we've talked multiple times about submitting PR without introducing distraction. Why do you submit those changes?

File: app/screens/main/Main.js [hunk 6]
```diff
@@ -231,20 +231,15 @@ class Main extends Component<Props, State> {
...
-  }
-
   componentDidUpdate(prevProps: Props) {
     const { isConnected, miningStatus, getMiningStatus, getGenesisTime } = this.props;
     if (isConnected && [nodeConsts.IN_SETUP, nodeConsts.IS_MINING].includes(miningStatus)) {
       getGenesisTime();
     }
     if (isConnected && prevProps.miningStatus === nodeConsts.NOT_MINING && miningStatus === nodeConsts.IN_SETUP) {
-      this.miningStatusInterval = setInterval(() => { isConnected && getMiningStatus(); }, 3600000);
+      this.miningStatusInterval = setInterval(() => {
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: same here

File: app/App.js [hunk 7]
```diff
@@ -122,12 +122,13 @@ class App extends React.Component<Props, State> {
     });
   };
 
-  healthCheckFlow = () => {
-    const healthCheckIntervalTime = 60000;
-    store.dispatch(checkNodeConnection());
-    this.healthCheckInterval = setInterval(() => {
-      store.dispatch(checkNodeConnection());
-    }, healthCheckIntervalTime);
+  connectionAndMiningFlow = async () => {
```
Comment (IlyaVi) [inline app/App.js]: entering this function means that isConnected === true? If it is then is it necessary to check connectivity once again?

File: app/redux/node/actions.js (hunk 2 above)
Comment (IlyaVi) [reply app/redux/node/actions.js]: After all reviews and changes made is this code still has to be changed?

File: app/screens/main/Main.js (hunk 5 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: Auto formatter does this... I didn't really change anything here.

Comment (IlyaVi) [reply app/screens/main/Main.js]: So you are not reviewing your PRs after submission, something we talked about a lot

File: app/screens/auth/CreateWallet.js [hunk 8]
```diff
@@ -192,10 +194,10 @@ class CreateWallet extends Component<Props, State> {
     return !passwordError && !verifyPasswordError;
   };
 
-  nextAction = () => {
-    const { history, location } = this.props;
+  nextAction = async () => {
```
Comment (IlyaVi) [inline app/screens/auth/CreateWallet.js]: remove async
